hcc.docker

```diff
@@ -0,0 +1,66 @@
+FROM ubuntu:18.04
+
+ARG PREFIX=/usr/local
+
+# Support multiarch
+RUN dpkg --add-architecture i386
+
+# Add rocm repository
+RUN sh -c 'echo deb [arch=amd64 trusted=yes] http://repo.radeon.com/rocm/apt/3.0/ xenial main > /etc/apt/sources.list.d/rocm.list'
+
+# Install dependencies
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --allow-unauthenticated \
+    apt-utils \
+    build-essential \
+    clang-format-5.0 \
+    cmake \
+    curl \
+    doxygen \
+    gdb \
+    git \
+    lcov \
+    pkg-config \
+    python \
+    python-dev \
+    python-pip \
+    python3 \
+    python3-dev \
+    python3-pip \
+    software-properties-common \
+    wget \
+    rocm-clang-ocl \
+    rocm-device-libs \
+    rocblas \
+    zlib1g-dev && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Workaround broken rocm packages
+RUN ln -s /opt/rocm-* /opt/rocm
+
+ENV LC_ALL=C.UTF-8
+ENV LANG=C.UTF-8
+
+# Install rbuild
+RUN pip3 install https://github.com/RadeonOpenCompute/rbuild/archive/master.tar.gz
+
+# Install doc requirements
+ADD doc/requirements.txt /doc-requirements.txt
+RUN pip3 install -r /doc-requirements.txt
+
+# Install dependencies
+ADD dev-requirements.txt /dev-requirements.txt
+ADD requirements.txt /requirements.txt
+# Manually ignore rocm dependencies
+RUN cget -p $PREFIX ignore \
+    RadeonOpenCompute/clang-ocl \
+    ROCm-Developer-Tools/HIP \
+    ROCmSoftwarePlatform/MIOpenGEMM \
```

Comment:
I am wondering why not install the miopen and rocblas using "apt install", this can avoid the long time used for building the source code...

Reply:
I do that already, which is why I ignore them here.

Reply:
Its installed by apt [here](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/pull/706/files/ec060537bbc0a9fd75226193da551f35d0e47fe1#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R35).

Reply:
Sorry, I think I had considered the miopendriver as the miopen when I try this docker file.